commands/cli.py

```diff
@@ -0,0 +1,35 @@
+from typing import Any
+
+from cleo.application import Application
+
+import commands
+from starlite import Starlite
+
+
+class Cli:
+    """CLI Add CLI commands using cleo https://github.com/python-poetry/cleo .
+    New commands should be placed in the commands folder and added in
+    __init__.py.
+
+    usage example:
+        # ...
+        app = Starlite(route_handlers=[hello_world])
+        if __name__ == '__main__':
+            from commands.cli import Cli
+            cli = Cli(app)
+            cli.run()
+    """
+
+    def __init__(self, starletapp: Starlite):
+        self.st_app = starletapp
+        self.cli = Application()
+        for command in commands.__all__:
+            cli_command = getattr(commands, command)
+            try:
+                self.cli.add(cli_command(self.st_app))
+            except Exception:
```

Comment:
whats happening here?

Reply:
It dynamically adds the commands, like   `cleo.add(my_command())`  ,  
however, some commands want access to the app,  like `cleo.add(ListRoutes(app))`  .

This tries both options, so it does not force anyone to have the `app` in their commands.  
I don't see another way to make this optional (but perhaps there is?).

Reply:
Yes, you can use inspection. Look at how we check if a lifespan handler requires state.